docs/theory/swe.md

```diff
@@ -0,0 +1,342 @@
+     - &\int_{x_1}^{x_2}&\mathbf{F}_y|_{y_1}~dx ~~~~\mathrm{(bottom)} \\
+     + &\int_{y_1}^{y_2}&\mathbf{F}_x|_{x_2}~dy ~~~~\mathrm{(right)} \\
+     + &\int_{x_1}^{x_2}&\mathbf{F}_y|_{y_2}~dx ~~~~\mathrm{(top)} \\
+     - &\int_{y_1}^{y_2}&\mathbf{F}_x|_{x_1}~dy ~~~~\mathrm{(left)}
+\end{eqnarray}
+
+Evidently $I_1 = I_2$ if $\mathbf{F}_x = \mathbf{E}$ and $\mathbf{F}_y = \mathbf{G}.$
+
+In the rest of this section, we use the flux form $(\ref{4})$ of the shallow
+water equations.
+
+### Finite volume formulation
+
+We can obtain a finite volume method for these equations by defining
+_horizontally-averaged_ quantities for flow depth and velocities:
+
+\begin{eqnarray}
+h_i &=& \frac{1}{A_i}\int_{\Omega_i} h d\Omega_i \\
+u_i &=& \frac{1}{A_i}\int_{\Omega_i} u d\Omega_i \\
+v_i &=& \frac{1}{A_i}\int_{\Omega_i} v d\Omega_i
+\end{eqnarray}
+
+where $A_i = \int_{\Omega_i}d\Omega_i$ is the area enclosed within cell $i$. We
+also introduce the _horizontally-averaged solution vector_
+
+$$
+\mathbf{U}_i = (h_i, h_i u_i, h_i v_i)^T
+$$
+
+and the _horizontally-averaged source vector_
+
+$$
+\mathbf{S}_i = \frac{1}{A_i}\int_{\Omega_i} \mathbf{S} d\Omega_i.
+$$
+
+Finally, we define the _face-averaged normal flux vector_ between cell $i$ and
+an adjoining cell $j$:
+
+$$
+\mathbf{F}_{ij} = \frac{1}{l_{ij}}\int_{\partial\Omega_i\bigcap\partial\Omega_j}\mathbf{\vec{F}}\cdot\vec{n}~dl \label{5}
+$$
+
+where $l_{ij}$ is the length of the face connecting cells $i$ and $j$.
+
+With these definitions, the shallow water equations in cell $i$ are
+
+$$
+\frac{\partial\mathbf{U}_i}{\partial t} + \sum_j\mathbf{F}_{ij} l_{ij} = \mathbf{S}_i, \label{6}
+$$
+
+where the index $j$ in each term of the sum refers to a neighboring cell of cell $i$.
+
+### Boundary conditions
+
+To incorporate boundary conditions, we partition the domain boundary $\Gamma$
+into disjoint line segments, each of which represents a _boundary face_
+$\Gamma_i$. Every cell $j$ that touches the boundary $\Gamma$ has at least one
+boundary face so that 
+
+$$
+\partial\Omega_j \bigcup \Gamma \neq \varnothing.
+$$
+
+Such a cell is a _boundary cell_. The boundary $\Gamma$ consists entirely of
+faces of boundary cells.
+
+In dealing with boundary conditions, we must distinguish between the faces a
+boundary cell $i$ _does_ and _does not_ share with the boundary $\Gamma$:
+
+$$
+\frac{\partial\mathbf{U}_i}{\partial t} +
+\sum_{j: \partial\Omega_j\subset\Gamma}\mathbf{F}_{ij}^{\Gamma} l_{ij} +
+\sum_{j: \partial\Omega_j\not\subset\Gamma}\mathbf{F}_{ij} l_{ij}
+= \mathbf{S}_i. \label{7}
+$$
+
+To enforce boundary conditions, we must compute the effective boundary fluxes
+$\mathbf{F}_{ij}^{\Gamma}$ that appear in the first sum in $(\ref{7})$. These
+boundary fluxes have specific forms depending on their respective boundary
+conditions.
+
+### Source terms
+
+The term for the vector-valued source for the water's momentum, which we
```

Comment:
How about we split the source term into three as we have done in the Overleaf document?

Reply:
I'll take care of this in my next pass.